### PR TITLE
Add a post-processor which outputs visually the depth according to the geometry model.

### PR DESCRIPTION
**Pull request by MFraters**
Add a post-processor which outputs visually the depth according to the geometry model. This is needed to visualise the problem in https://github.com/geodynamics/aspect/issues/643. I don't think making a test will be necessary for this, but if it is, let me know.

### CODE REVIEW

**Comment by bangerth** [comment]
I actually like this feature because it can be used in Visit or Paraview to select a certain part of the domain at a certain depth. @dsarahstamps and I tried to strenously do this in different ways, but it will be much easier to use the data output here as a selector of a part of the domain (a subdomain, or just an isosurface at a certain depth) where to show some other quantity. I have no idea why I didn't think of writing this postprocessor myself at the time -- nicely done.

So yes, this is cool. I'd really like to have a test, though, if we keep this for later. Would you mind adding one, and also adding an entry to `changes.h`?

**Comment by MFraters** [comment]
It is good to hear that it is useful for more than just the what I had in mind, which was to help with debugging geometry models :)

I will add a simple test to test this plugin with a box geometry (I could also add a test for every geometry model to test if the depth calculation has stayed the same) and an entry to changes.h.

File: source/postprocess/visualization/depth.cc [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(Depth,
+                                                  "depth",
+                                                  "A visualization output object that outputs the depth according to the geometry model.")
```
Comment (bangerth) [inline source/postprocess/visualization/depth.cc]: Can you say "the depth for all points inside the domain, as determined by the geometry model"?

**Comment by bangerth** [comment]
I think one test is ok with me. Having one for each geometry model would of course be even better (there can never be enough tests) if you feel ambitious! :-)

**Comment by tjhei** [comment]
Looks good but please add one test.

Please don't add 50 new tests even if Wolfgang seems to like the idea. :-)

**Comment by bangerth** [comment]
> Please don't add 50 new tests even if Wolfgang seems to like the idea. :-)
:-) I was actually only half joking. We have only 6 geometry models, and I 
wouldn't be surprised if the `depth()` function is not well tested ;-)

**Comment by MFraters** [comment]
I will make those 6 tests really small :)

**Comment by MFraters** [comment]
I have to wait for the ellipsoidal fixes pull request, to make the ellipsoidal test.

**Comment by MFraters** [comment]
I added the ellipsoidal geometry model to the tests, so I think this can also be merged now if there are no further comments.

**Comment by bangerth** [comment]
Thanks!

**Pull request merged**